pylint/constants.py

```diff
@@ -54,6 +56,51 @@
 DEFAULT_PYLINT_HOME = platformdirs.user_cache_dir("pylint")
 
 
+def get_pylint_home() -> str:
+    """Return the pylint home."""
+    if "PYLINTHOME" in os.environ:
+        return os.environ["PYLINTHOME"]
+    pylint_home = DEFAULT_PYLINT_HOME
+    # The spam prevention is due to pylint being used in parallel by
+    # pre-commit, and the message being spammy in this context
+    # Also if you work with old version of pylint that recreate the
+    # old pylint home, you can get the old message for a long time.
+    prefix_spam_prevention = "pylint_warned_about_old_cache_already"
+    spam_prevention_file = os.path.join(
+        pylint_home,
+        datetime.now().strftime(prefix_spam_prevention + "_%Y-%m-%d.temp"),
+    )
+    old_home = os.path.join(USER_HOME, OLD_DEFAULT_PYLINT_HOME)
+    if os.path.exists(old_home) and not os.path.exists(spam_prevention_file):
```

Comment:
Do we want to test this?

Reply:
I think so ? It probably means passing the parameter in the function.